File: hw/armijos/u3/hw30StrategyandObserver/StrategyPattern/src/main/java/ec/edu/espe/sortingapp/controller/InsertionSort.java
package ec.edu.espe.sortingapp.controller;

/**
 *
 * @author Code by Shakhzod Yuld https://github.com/soloShak/Java-Sorting-app.git
 */
public class InsertionSort implements SortingStrategy{

    @Override
    public int[] sort(int[] data) {
        int n = data.length;
        for (int i = 1; i < n; ++i) {
            int key = data[i];
            int j = i - 1;

            while ((j >= 0) && (data[j] > key)) {
                data[j + 1] = data[j];
                j--;
            }
            data[j + 1] = key;
        }
        return data;
    }

}

File: exams/pantoja/u3/Strategy/src/main/java/ec/espe/edu/strategy/model/InsertionSort.py
class InsertionSort:
    
    def sort(self, data):
        print("\tInsertion Sort")
        n = len(data)
        for i in range(1, n):
            key = data[i]
            j = i - 1

            while j >= 0 and data[j] > key:
                data[j + 1] = data[j]
                j -= 1

            data[j + 1] = key
        return data

